### PR TITLE
Show info for Metric Details Page

### PR DESCRIPTION
**Pull request by meghajain-1711**
Added Metric Info to the Metric page.

The keys currently are the same as media_state_play page ( bugs, data_reviews, lifetime, disabled, .. ) 

Formatting/styling has not yet been added as mentioned in the Issue. But the code has been written so that it can be formatted similar to the above mentioned page ( Check the image in Issue) 

More info (help) has been linked for lifetime key. It can be further added for others in the similar manner.

This is for Issue : Metric pages show almost no info #66

### CODE REVIEW

**Comment by meghajain-1711** [comment]
Hi @wlach This PR is ready for review :)

**Review by wlach — CHANGES_REQUESTED** [review]
Thanks for the PR! It looks like CI is failing, can you take a look at why? It's probably something simple. If it's a prettier/formatting problem, try running `npm run format`.

After that's fixed, I can take a more detailed look later.

**Comment by meghajain-1711** [comment]
> Thanks for the PR! It looks like CI is failing, can you take a look at why? It's probably something simple. If it's a prettier/formatting problem, try running `npm run format`.
> 
> After that's fixed, I can take a more detailed look later.

Hi @wlach Thanks ! It was a prettier formatting issue only :) Had to update my npm intsallation. Also, running npm test after npm run format worked.  Kindly review whenever possible.

**Comment by meghajain-1711** [comment]
1. The keys can be further extended to look at all the keys in the json
2. Currently formatting is with footer and div since I had almost implemented a similar to media_play on click tab to open. But removed that formatting to keep it consistent with the mentioned issue. 
3. Help/More info as mentioned has been added only for lifetime. If that seems correct, I can extend it to other keys as well. Or if needed we can tackle it in another issue. 

Your comments are welcome :)

File: src/pages/MetricDetail.svelte [hunk 1]
```diff
@@ -2,10 +2,116 @@
...
   export let params;
+
   const metricDataPromise = getMetricData(params.app, params.metric);
 </script>
 
 {#await metricDataPromise then metric}
   <h1>{metric.name}</h1>
-  <p>{metric.description}</p>
+  <h2>{metric.description}</h2>
+  <div />
```
Comment (wlach) [inline src/pages/MetricDetail.svelte:12]: No need for an empty div here.

```suggestion
```

File: src/pages/MetricDetail.svelte [hunk 2]
```diff
@@ -2,10 +2,116 @@
...
+  <section id="metric-detail-view">
+    <div>
+      <footer>
+        <div>
+          <div>
```
Comment (wlach) [inline src/pages/MetricDetail.svelte:17]: Likewise, most of these elements are unnecessary.

```suggestion
```

File: src/pages/MetricDetail.svelte [hunk 3]
```diff
@@ -2,10 +2,116 @@
...
+            <dl>
```
Comment (wlach) [inline src/pages/MetricDetail.svelte:18]: If we're going to use a description list here, we would want something like:

```svelte
<dl>
<dt>Relevant Bugs</dt>
{#each metrics.bugs as bug, i}
 <dd><a href={bug} ...>{i+1}</a></dd>
```

For more information see: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dl

File: src/pages/MetricDetail.svelte [hunk 4]
```diff
@@ -2,10 +2,116 @@
...
 
   export let params;
+
   const metricDataPromise = getMetricData(params.app, params.metric);
 </script>
 
 {#await metricDataPromise then metric}
   <h1>{metric.name}</h1>
-  <p>{metric.description}</p>
+  <h2>{metric.description}</h2>
```
Comment (wlach) [inline src/pages/MetricDetail.svelte:11]: This should not be a header, the previous paragraph was correct.

```suggestion
  <p>{metric.description}</p>
```

**Review by wlach — CHANGES_REQUESTED** [review]
@meghajain-1711 great start! I have some concerns about the HTML-- there seem to be some elements that are unnecessary in your PR. If they were added for formatting purposes, don't worry about that for now-- our goal at this point is just to get the scaffolding right (i.e. using approximately the right set of html elements to represent our data), we'll be adding more CSS shortly.

File: src/pages/MetricDetail.svelte (hunk 3 above)
Comment (meghajain-1711) [reply src/pages/MetricDetail.svelte]: Yes ! I was trying to do the same :) Description List was trying to keep inline with older netlify app. But given that all the formatting had not been added, I have minimalized it to a simple Table. 
Thanks !

**Comment by meghajain-1711** [comment]
Hi @wlach 
I have removed all empty divs and reformatted it to a simpler table format. As you said we can add styling later.
Would be looking like this. 
Any comments are welcome :)

image

File: src/pages/MetricDetail.svelte [hunk 5]
```diff
@@ -2,10 +2,126 @@
...
+      <tr>
+        <td>Range Maximum</td>
+        <td>{metric.range_max}</td>
+      </tr>
+    {/if}
+    <tr>
+      <td>Disabled</td>
+      <td>{metric.disabled}</td>
+    </tr>
+    {#if metric.data_reviews && metric.data_reviews.length}
```
Comment (wlach) [inline src/pages/MetricDetail.svelte:96]: We can assume there is a data reviews property, it's required by the schema:

https://github.com/mozilla/glean_parser/blob/5102e037a12924fa8585dd9d4f303d41027936f1/glean_parser/schemas/metrics.1-0-0.schema.yaml#L475

File: src/pages/MetricDetail.svelte [hunk 6]
```diff
@@ -2,10 +2,126 @@
...
+          {/each}
+        </td>
+      </tr>
+    {/if}
+    {#if metric.no_lint && metric.no_lint.length}
+      <tr>
+        <td>No_lint</td>
+        <td>{metric.no_lint.join(', ')}</td>
+      </tr>
+    {/if}
```
Comment (wlach) [inline src/pages/MetricDetail.svelte:113]: Let's skip this property, it seems to be an internal-only thing for the parser, probably not very interesting for a consumer

```suggestion
```

File: src/pages/MetricDetail.svelte [hunk 7]
```diff
@@ -2,10 +2,126 @@
...
+    {#if metric.labels && metric.labels.length}
+      <tr>
+        <td>Labels</td>
+        <td>{metric.labels.join(', ')}</td>
+      </tr>
+    {/if}
+    {#if metric.version || metric.version === 0}
```
Comment (wlach) [inline src/pages/MetricDetail.svelte:120]: We should always show something here, it should be `0` if there is no version:

https://github.com/mozilla/glean_parser/blob/5102e037a12924fa8585dd9d4f303d41027936f1/glean_parser/schemas/metrics.1-0-0.schema.yaml#L258

File: src/pages/MetricDetail.svelte [hunk 8]
```diff
@@ -2,10 +2,126 @@
   import { getMetricData } from "../state/api";
 
   export let params;
+
   const metricDataPromise = getMetricData(params.app, params.metric);
 </script>
 
+<style>
+  .table-header {
```
Comment (wlach) [inline src/pages/MetricDetail.svelte:10]: Let's call this "metrics-table" or something, as it's not really a header.

```suggestion
  .metrics-table {
```

File: src/pages/MetricDetail.svelte [hunk 9]
```diff
@@ -2,10 +2,126 @@
...
+        {#each metric.bugs as bug, i}
+          {#if bug.indexOf('http') > -1}
+            <a href={bug} title={bug} target="_blank"> {i + 1} </a>
+          {:else}<span>{bug}</span>{/if}
+        {/each}
+      </td>
+    </tr>
+    <tr>
+      <td>Send In Pings</td>
+      <td>{metric.send_in_pings.join(', ')}</td>
```
Comment (wlach) [inline src/pages/MetricDetail.svelte:36]: Can you make this a link to the actual ping views? e.g. http://localhost:5000/#!/apps/burnham/pings/discovery

**Review by wlach — CHANGES_REQUESTED** [review]
Getting close! There's a few things I'd like to address before we merge.

File: src/pages/MetricDetail.svelte (hunk 9 above)
Comment (meghajain-1711) [reply src/pages/MetricDetail.svelte]: Here , for few metrics under fenix application like "about_page.libraries_tapped" the ping value is "events" 
But I dont see fenix/pings/events as a valid json
Nor do I see "events" in the list of pings under fenix : http://localhost:5000/data/fenix/tables/events.json is nonexistent

Please guide me on what I might be missing here :)
Thanks.

**Comment by meghajain-1711** [comment]
Hi @wlach This PR is ready for review :) 

Along with that, whenever possible, kindly check if  this PR [https://github.com/mozilla/glean-dictionary/pull/103](url) can be merged here as well :)

File: src/pages/MetricDetail.svelte (hunk 9 above)
Comment (wlach) [reply src/pages/MetricDetail.svelte]: You actually stumbled on a missing feature: we're not processing dependencies for our applications. See (just filed) #51 

For now you can test with this metric:

http://localhost:5000/#!/apps/fenix/metrics/activation.activation_id

The found in pings should have the activation ping, and the URL you provide should take you to:

http://localhost:5000/#!/apps/fenix/pings/activation

File: src/pages/MetricDetail.svelte [hunk 10]
```diff
@@ -2,10 +2,123 @@
...
+      <td>Disabled</td>
+      <td>{metric.disabled}</td>
+    </tr>
+    <tr>
+      <td>Data Reviews</td>
+      <td>
+        {#each metric.data_reviews as rev, i}
+          {#if rev.indexOf('http') > -1}
+            <a href={rev} title={rev} target="_blank"> {i + 1} </a>
+          {:else}<span>{rev}</span>{/if}
```
Comment (wlach) [inline src/pages/MetricDetail.svelte:105]: We can assume it's a bugzilla link if it isn't specified as a full URL.

```suggestion
          {:else}<a href="https://bugzilla.mozilla.org/show_bug.cgi?id={rev}">{i+1}</a>{/if}
```

**Review by wlach — CHANGES_REQUESTED** [review]
Thanks for your persistence on this! A few last issues, thanks for your persistence.

**Comment by meghajain-1711** [comment]
> Thanks for your persistence on this! A few last issues.

Thanks for your continuous review effort on this @wlach !

**Review by wlach — CHANGES_REQUESTED** [review]
Looks like there's a merge conflict with your previous PR. After that's resolved, I'm happy to merge.

**Comment by meghajain-1711** [comment]
Hi @wlach ! All conflicts resolved :) Should be good for checkin now ! 
Our final Metric Page looks like this now : 
image

**Review by wlach — CHANGES_REQUESTED** [review]
Hi! On testing I found one other problem, please address and then we can merge.

**Comment by meghajain-1711** [comment]
Hey @wlach , Thanks for that incredible insight. I am so sorry to have missed this. 
I have updated the PR with the requested change. 

Thanks again fr the review :) 

Kindly look into it whenever possible !

**Review by wlach — APPROVED** [review]

**Comment by wlach** [comment]
Thank you!

**Pull request merged**